pkg/lib/config/saml.go

```diff
@@ -40,3 +42,55 @@ func (c *SAMLConfig) ResolveProvider(id string) (*SAMLServiceProviderConfig, boo
 type SAMLServiceProviderConfig struct {
 	ID string `json:"id,omitempty"`
 }
+
+var _ = Schema.Add("SAMLSigningConfig", `
+{
+	"type": "object",
+	"additionalProperties": false,
+	"properties": {
+		"key_id": { "type": "string" },
+		"signature_method": { "$ref": "#/$defs/SAMLSigningSignatureMethod" },
+		"digest_method": { "$ref": "#/$defs/SAMLSigningDigestMethod" }
+	}
+}
+`)
+
+type SAMLSigningConfig struct {
+	KeyID           string                     `json:"key_id,omitempty"`
+	SignatureMethod SAMLSigningSignatureMethod `json:"signature_method,omitempty"`
+	DigestMethod    SAMLSigningDigestMethod    `json:"digest_method,omitempty"`
+}
+
+func (c *SAMLSigningConfig) SetDefaults() {
+	if c.SignatureMethod == "" {
+		c.SignatureMethod = SAMLSigningSignatureMethodRSASHA256
+	}
+	if c.DigestMethod == "" {
+		c.DigestMethod = SAMLSigningDigestMethodSHA256
+	}
+}
+
+var _ = Schema.Add("SAMLSigningSignatureMethod", `
+{
+	"enum": ["RSAwithSHA256"] 
+}
+`)
+
+type SAMLSigningSignatureMethod string
+
+const (
+	SAMLSigningSignatureMethodRSASHA256 SAMLSigningSignatureMethod = "RSAwithSHA256"
+)
+
+var _ = Schema.Add("SAMLSigningDigestMethod", `
+{
+	"enum": ["SHA256", "SHA1"] 
```

Comment:
Support SHA256 only. Remove the support for SHA1

Reply:
It seems SHA1 is still widely supported as digest method in different implementations, but I think we can add it back easily if necessary so remove first.